src/search.jl

```diff
@@ -36,21 +36,30 @@ function Base.searchsorted(a::Range, I::ClosedInterval)
     searchsortedfirst(a, I.left):searchsortedlast(a, I.right)
 end
 
-if VERSION > v"0.5.0-dev+4557"
-    # When running with "--check-bounds=yes" (like on Travis), the bounds-check isn't elided
-    @inline function Base.unsafe_getindex{T}(v::Range{T}, i::Integer)
-        convert(T, first(v) + (i-1)*step(v))
-    end
+# When running with "--check-bounds=yes" (like on Travis), the bounds-check isn't elided
+@inline function Base.unsafe_getindex{T}(v::Range{T}, i::Integer)
```

Comment:
The point of JuliaLang/julia#14957 was to centralize/simplify, but for ranges there do turn out to be cases where you want to perform the computation even when bounds-checking is forced on. Since a certain number of `unsafe_getindex` calls have crept back into range.jl, maybe it's not unreasonable to add them back systematically.